taf/testlib/virtual_env.py

```diff
@@ -310,17 +376,19 @@ def ensure_public_access(self, try_reuse=False, networks_client=None, routers_cl
 
         _net_cfg.public_network_id = public_network['id']
 
-        router_kwargs = {
-            'routers_client': routers_client,
-            'network_id': _net_cfg.public_network_id,
-            'tenant_id': tenant_id,
-            'enable_snat': True
-        }
-        public_router = self.create_router(**router_kwargs)
-        assert public_router
-        _net_cfg.public_router_id = public_router['id']
+        if create_external_router:
+            router_kwargs = {
+                'routers_client': routers_client,
+                'network_id': _net_cfg.public_network_id,
+                'tenant_id': tenant_id,
+                'enable_snat': True
+            }
+            public_router = self.create_router(**router_kwargs)
+            assert public_router
+            _net_cfg.public_router_id = public_router['id']
 
-        if _net_cfg.public_network_id and _net_cfg.public_router_id:
+        if _net_cfg.public_network_id and (_net_cfg.public_router_id or
```

Comment:
Please return the result of the Boolean expression instead of testing the Boolean expression and returning a Boolean literal.

```
return _net_cfg.public_network_id and (...)
```